jupyter/nmm-gui.ipynb

```diff
@@ -1244,7 +1297,11 @@
    "outputs": [],
    "source": [
     "def checkIfFinished(self):\n",
-    "    if finished(self.state, self.player):\n",
+    "    if (self.limitMovesWithoutMill is not None) and (self.movesWithoutMill >= self.limitMovesWithoutMill):\n",
+    "        self.winner = NO_PLAYER\n",
+    "    elif (self.limitStatesCounter is not None) and (self.statesCounter[hash(self.state)] >= self.limitStatesCounter):\n",
```

Comment:
Welchen Vorteil gewinnen wir dadurch, dass die States gehasht werden? Mir fällt gerade nur die Speicherreduktion ein, aber da eh nicht so viele States gespeichert werden (vielleicht 100 oder 200 maximal in einem Spiel), macht es doch keinen Unterschied, oder?

Reply:
ich dachte ehrlich gesagt, dass man tupel nicht als key speichern kann xD

aber its python magic
![](https://media1.tenor.com/images/00ba7c39e7cdc7881773a32ef532ef62/tenor.gif?itemid=18115910)